src/LightSm.example.test.c

```diff
@@ -0,0 +1,57 @@
+// This is a sample test file for the LightSm state machine.
+// Generated by StateSmith.
+// Feel free to make a copy and start adding your own tests.
+
+// This sample uses CppUTest to run tests.
+//
+// To run the tests:
+// - install CppUTest if you don't already have it (https://cpputest.github.io/)
+// - make a copy of this file, eg. LightSm.test.c
+// - mock any dependencies from your state machine (see ACTION REQUIRED below)
+// - compile your state machine and tests, eg.
+//   g++ LightSm.test.c LightSm.c -L/path/to/CppUTest/lib/dir -lCppUTest
+// - make the tests executable, eg.
+//   chmod +x a.out
+// - run the tests, eg.
+//   ./a.out
+
+#include "CppUTest/CommandLineTestRunner.h"
+#include "LightSm.h"
+// ACTION REQUIRED
+// Create LightSm.mocks.test.h and add your mocks there.
+//
+// You must define any actions and variables used by your state machine.
+// We recommend mocking rather than importing your actual functions,
+// to keep these tests purely about testing the state machine itself.
+// (Your function implementations should also be tested, but in separate tests.)
+
+#include "LightSm.mocks.test.h"
+
+
+int main(int ac, char** av) {
```

Comment:
The problem with adding a main function is that it prevents running a bunch of test modules together. They all have their own main function and the linker will see multiple main function symbols.

For example, I will often have a test directory that has a number of unit test files. It might look like this:
```
src/
├─ main.c
├─ LightSm.c/h
├─ RocketSm.c/h
test/
├─ main.cpp
├─ LightSm.test.cpp
├─ RocketSm.test.cpp
```

The test code could live beside the production code but it often doesn't at my work. It makes certain things handy like being able to do `g++ test/*`

Could we add a comment saying to delete the main function if another already exists?